src/controllers/userLeaderboard.controller/getUserLeaderboard/index.js

```diff
@@ -0,0 +1,34 @@
+const { validationResult } = require('express-validator');
+const { query: Hasura } = require('../../../utils/hasura');
+const { getUserLeaderboardQuery } = require('./queries/queries');
+const catchAsync = require('../../../utils/catchAsync');
+
+const getUserLeaderboard = catchAsync(async (req, res) => {
+  const sanitizerErrors = validationResult(req);
+  if (!sanitizerErrors.isEmpty()) {
+    return res.status(400).json({
+      success: false,
+      ...sanitizerErrors,
+    });
+  }
+
+  const { cognito_sub } = req.body;
+
+  const response = await Hasura(getUserLeaderboardQuery, {
```

Comment:
Let's implement pagination for all new fetch APIs, starting from this one. The client should be able to send page size & page number, we can use limit & offset in the graphql query to return only the records of that page